### PR TITLE
Remove obsolete v2-v5 record stream file migration code

### LINKED ISSUE #1420: Remove obsolete V2 -> V5 record migration (causes problem in blank-state restarts)
**Summary of the defect**
Although we have no environments with V2 record stream files, `ServiceState` still tries to re-initialize an empty `runningHashLeaf` on restart from any records, _assuming_ these will be V3 records. 

Of course the records are actually V5 (if present), and this causes parsing errors -> restart fails.

### PR DESCRIPTION
**Pull request by anighanta**
**Related issue(s)**:
Closes #1420 

**Summary of the change**:
Removed v2-v5migration flow as we will no longer use the v2 files even during migration.
Safely removed class `RecordStream` 

**External impacts**:
None.

### CODE REVIEW

**Review by Neeharika-Sompalli — APPROVED** [review]
Please run ND correctness reconnect tests and attach the passing test link to the PR.

**Review by qnswirlds — APPROVED** [review]

File: hedera-node/src/main/java/com/hedera/services/ServicesState.java [hunk 1]
```diff
@@ -237,6 +235,11 @@ public void init(Platform platform, AddressBook addressBook) {
 			setChild(ChildIndices.TOKEN_ASSOCIATIONS, new FCMap<>());
 			setChild(ChildIndices.DISK_FS, new MerkleDiskFs());
 			setChild(ChildIndices.SCHEDULE_TXS, new FCMap<>());
+
+			/* Initialize the running hash leaf at genesis to an empty hash. */
+			final var firstRunningHash = new RunningHash();
+			firstRunningHash.setHash(emptyHash);
+			setChild(ChildIndices.RECORD_STREAM_RUNNING_HASH, new RecordsRunningHashLeaf(firstRunningHash));
```
Comment (ljianghedera) [inline hedera-node/src/main/java/com/hedera/services/ServicesState.java:242]: Is this section for migration from older version? Is so, will the initialization to empty hash cause some information loss?

File: hedera-node/src/main/java/com/hedera/services/state/exports/SigFileWriter.java [hunk 2]
```diff
@@ -22,5 +22,9 @@
 
 @FunctionalInterface
 public interface SigFileWriter {
+
+	byte TYPE_SIGNATURE = 3;       // the file content signature, should not be hashed
+	byte TYPE_FILE_HASH = 4;       // next 48 bytes are hash384 of content of the file to be signed
```
Comment (abhishek-hedera) [inline hedera-node/src/main/java/com/hedera/services/state/exports/SigFileWriter.java:27]: Wouldn't it be better to make them `final` too?

Comment (anighanta) [reply hedera-node/src/main/java/com/hedera/services/state/exports/SigFileWriter.java:27]: we don't have to use `final` in interfaces

**Pull request merged**